### PR TITLE
[Keyboard] Add Ano

### PR DESCRIPTION
**Pull request by sauvehoo**
Adding custom keyboard.

## Description

18x6 PCB build with anodized aluminum plate and wooden base.  Rotary encoder and numpad included with some custom layout changes.
 
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ano/config.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ 
+ #pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x8372
+#define PRODUCT_ID      0x0651
+#define DEVICE_VER      0x0000
+#define MANUFACTURER    Sebastien Sauve-Hoover
+#define PRODUCT         Ano Keyboard
+#define DESCRIPTION     Custom made keyboard
```
Comment (drashna) [inline keyboards/ano/config.h:25]:
```suggestion
```

File: keyboards/ano/config.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+#define DIODE_DIRECTION COL2ROW
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE    0
+
+#define TAPPING_TERM 200
```
Comment (drashna) [inline keyboards/ano/config.h:47]:
```suggestion
```

File: keyboards/ano/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+         Page Down key. You can do anything QMK allows you to do here.
+         You'll want to replace these lines with the things you want your
+         encoders to do. */
+      tap_code(KC_AUDIO_VOL_UP);
+    } else {
+      /* And likewise for the other direction, this time Page Down is pressed. */
+      tap_code(KC_AUDIO_VOL_DOWN);
+    }
+  } 
+}
```
Comment (drashna) [inline keyboards/ano/keymaps/default/keymap.c:64]:
```suggestion
```

File: keyboards/ano/rules.mk [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# MCU name
+MCU = STM32F303
+ENCODER_ENABLE = yes
+EXTRAKEY_ENABLE = yes
```
Comment (drashna) [inline keyboards/ano/rules.mk:4]:
```suggestion
# MCU name
MCU = STM32F303

# Bootloader selection
BOOTLOADER = stm32-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]
Needs a readme, as well.

File: keyboards/ano/ano.c [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#include "ano.h"
```
Comment (drashna) [inline keyboards/ano/ano.c:17]:
```suggestion
#include "ano.h"

#ifdef ENCODER_ENABLE
void encoder_update_user(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) { 
        if (clockwise) {
            tap_code(KC_AUDIO_VOL_UP);
        } else {
            tap_code(KC_AUDIO_VOL_DOWN);
        }
    }
    return true; 
}
#endif
```

File: keyboards/ano/ano.c [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * The index parameter tells you which encoder was turned. If you only have
+ * one encoder, the index will always be zero.
+ * 
+ * The clockwise parameter tells you the direction of the encoder. It'll be
+ * true when you turned the encoder clockwise, and false otherwise.
+ */
+ 
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/ano/ano.c:30]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ano/ano.c (hunk 6 above)
Comment (sauvehoo) [reply keyboards/ano/ano.c:30]: Any reason to this change? I took  "encoder_update_user" directly from https://docs.qmk.fm/#/feature_encoders?id=encoders.

Comment (drashna) [reply keyboards/ano/ano.c:30]: the user function is for keymaps, the kb function is for the keyboard.

And this setup allows you to set a default behavior that can be overridden by the user function.

File: keyboards/ano/readme.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# ano
+
+95 key custom keyboard made as DIY project
+
+* Keyboard Maintainer: [keebnewb](https://github.com/sauvehoo)
+* Hardware Supported: Proton C
+* Hardware Availability: Not available
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/ano/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ano/readme.md (hunk 7 above)
Comment (sauvehoo) [reply keyboards/ano/readme.md:9]: Added information on how to enter bootloader.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Missing info.json.

File: keyboards/ano/ano.h [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_ano_6x18( \
```
Comment (noroadsleft) [inline keyboards/ano/ano.h:21]: If this macro supports every possible switch location, this should be named:

```suggestion
#define LAYOUT_all( \
```

File: keyboards/ano/info.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{
+    "keyboard_name": "Ano", 
+    "url": "https://github.com/sauvehoo", 
+    "maintainer": "sauvehoo", 
+    "layouts": {
+        "LAYOUT_ano_6x18": {
```
Comment (noroadsleft) [inline keyboards/ano/info.json:6]: This must be a case-sensitive match for the name used in `ano.h`.

```suggestion
        "LAYOUT_all": {
```

It's also missing two key objects, but let's deal with that later.

File: keyboards/ano/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ #include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { 
+
+  [0] = LAYOUT_ano_6x18(
```
Comment (noroadsleft) [inline keyboards/ano/keymaps/default/keymap.c:21]:
```suggestion
  [0] = LAYOUT_all(
```

File: keyboards/ano/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+	KC_ESC,	KC_F1,	KC_F2, 	KC_F3, 	KC_F4,	KC_F5,	KC_F6, 	KC_F7, 	KC_F8,	KC_F9, 	KC_F10, KC_F11, KC_F12, 	KC_DEL,         		        		 
+	KC_GRV,	KC_1,	KC_2,	KC_3, 	KC_4, 	KC_5, 	KC_6, 	KC_7, 	KC_8, 	KC_9, 	KC_0, 	KC_MINS,KC_EQL, 	KC_BSPC,		KC_NLCK,KC_PSLS,KC_PAST,KC_MEDIA_PLAY_PAUSE, 
+	KC_TAB,	KC_Q, 	KC_W, 	KC_E, 	KC_R, 	KC_T, 	KC_Y, 	KC_U, 	KC_I, 	KC_O, 	KC_P, 	KC_LBRC,KC_RBRC,	KC_BSLS,		KC_P7, 	KC_P8, 	KC_P9, 	KC_B, 
+	KC_CAPS,KC_A, 	KC_S, 	KC_D, 	KC_F, 	KC_G, 	KC_H, 	KC_J, 	KC_K, 	KC_L, 	KC_SCLN,KC_QUOT,			KC_ENT,			KC_P4, 	KC_P5, 	KC_P6, 	KC_PMNS, 
+	KC_LSFT,       	KC_Z, 	KC_X, 	KC_C, 	KC_V, 	KC_B, 	KC_N, 	KC_M, 	KC_COMM,KC_DOT,	KC_SLSH,KC_RSFT,	KC_UP,			KC_P1, 	KC_P2, 	KC_P3,  KC_PPLS,     
+	KC_LCTL,KC_LGUI,KC_LALT,   		                KC_SPC,         		        MO(1),	KC_RCTL,KC_LEFT,	KC_DOWN,		KC_RGHT,KC_P0, 	KC_PDOT,KC_PENT
+   ),
+	
+  [1] = LAYOUT_ano_6x18(
```
Comment (noroadsleft) [inline keyboards/ano/keymaps/default/keymap.c:30]:
```suggestion
  [1] = LAYOUT_all(
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/ano/config.h [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/ano/config.h:67]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**